eos-updater/poll-common.c

```diff
@@ -407,20 +380,77 @@ cleanstr (gchar *s)
 }
 
-EosMetricsInfo *
-eos_metrics_info_new (const gchar *booted_ref)
+static GVariant *
+checkpoint_blocked_event_payload_new (const char         *booted_ref,
+                                      const char         *target_ref,
+                                      EuuCheckpointBlock  reason)
 {
   g_autoptr(GHashTable) hw_descriptors = NULL;
-  g_autoptr(EosMetricsInfo) info = NULL;
+  g_autofree gchar *vendor, *product;
+  const char *reason_nick;
 
   hw_descriptors = get_hw_descriptors ();
 
-  info = g_object_new (EOS_TYPE_METRICS_INFO, NULL);
-  info->vendor = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, VENDOR_KEY)));
-  info->product = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, PRODUCT_KEY)));
-  info->ref = g_strdup (booted_ref);
+  vendor = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, VENDOR_KEY)));
+  product = cleanstr (g_strdup (g_hash_table_lookup (hw_descriptors, PRODUCT_KEY)));
+
+  reason_nick = euu_checkpoint_block_to_string (reason);
+
+  return g_variant_new ("(sssss)", /* snake noises are valid type signatures */
+                        vendor,
+                        product,
+                        booted_ref,
+                        target_ref,
+                        reason_nick ?: "");
+}
+
+static gboolean
+stop_aggregate_timer (gpointer data)
+{
+  g_return_val_if_fail (EMTR_IS_AGGREGATE_TIMER (data), G_SOURCE_REMOVE);
+
+  emtr_aggregate_timer_stop (data);
+  return G_SOURCE_REMOVE;
+}
+
+static void
+maybe_send_checkpoint_blocked_event (GVariant *payload)
+{
+#ifdef HAS_EOSMETRICS_0
+  g_return_if_fail (payload != NULL);
+  g_autofree gchar *payload_str = g_variant_print (payload, FALSE);
 
-  return g_steal_pointer (&info);
+  if (euu_get_metrics_enabled ())
+    {
+      EmtrAggregateTimer *timer;
+
+      g_message ("Recording metric event %s: %s",
+                 EOS_UPDATER_METRIC_CHECKPOINT_BLOCKED,
+                 payload_str);
+
+      /* What we actually want to do is increment the counter by 1. There is no
+       * API for this; or rather, there is a RecordAggregateEvent D-Bus API, but
+       * it is a no-op. https://phabricator.endlessm.com/T33511
+       *
+       * Fake it by using the API we do have: start a timer, and stop it after
+       * approximately one second.
+       */
+      timer = emtr_event_recorder_start_aggregate_timer (emtr_event_recorder_get_default (),
+                                                         EOS_UPDATER_METRIC_CHECKPOINT_BLOCKED,
+                                                         payload);
+      g_timeout_add_seconds_full (G_PRIORITY_DEFAULT,
+                                  1,
+                                  stop_aggregate_timer,
+                                  timer,
+                                  g_object_unref);
+    }
+  else
+    {
+      g_message ("Skipping metric event %s: %s (metrics disabled)",
+               EOS_UPDATER_METRIC_CHECKPOINT_BLOCKED,
+               payload_str);
```

Comment:
I presume this was `g_debug` before and now the alignment is off.

Reply:
Yes, I changed this for debugging purposes and didn't mean to commit it.